modules/alloy/src/main/ts/ephox/alloy/keying/SpecialType.ts

```diff
@@ -45,12 +45,13 @@ const getKeydownRules = (component: AlloyComponent, simulatedEvent: NativeSimula
   KeyRules.rule(KeyMatch.inSet(Keys.DOWN), specialInfo.onDown),
   KeyRules.rule(KeyMatch.inSet(Keys.LEFT), specialInfo.onLeft),
   KeyRules.rule(KeyMatch.inSet(Keys.RIGHT), specialInfo.onRight),
-  KeyRules.rule(KeyMatch.inSet(Keys.SPACE), specialInfo.onSpace),
-  KeyRules.rule(KeyMatch.inSet(Keys.ESCAPE), specialInfo.onEscape)
+  KeyRules.rule(KeyMatch.inSet(Keys.SPACE), specialInfo.onSpace)
 ];
 
-const getKeyupRules = (component: AlloyComponent, simulatedEvent: NativeSimulatedEvent, specialInfo: SpecialConfig): Array<KeyRules.KeyRule<SpecialConfig, Stateless>> => specialInfo.stopSpaceKeyup ? [
+const getKeyupRules = (component: AlloyComponent, simulatedEvent: NativeSimulatedEvent, specialInfo: SpecialConfig): Array<KeyRules.KeyRule<SpecialConfig, Stateless>> => (specialInfo.stopSpaceKeyup ? [
   KeyRules.rule(KeyMatch.inSet(Keys.SPACE), stopEventForFirefox)
-] : [ ];
+] : [ ]).concat([
+  KeyRules.rule(KeyMatch.inSet(Keys.ESCAPE), specialInfo.onEscape)
+]);
```

Comment:
This might be a little more readable?
```suggestion
const getKeyupRules = (component: AlloyComponent, simulatedEvent: NativeSimulatedEvent, specialInfo: SpecialConfig): Array<KeyRules.KeyRule<SpecialConfig, Stateless>> => [
  ...(specialInfo.stopSpaceKeyup ? [ KeyRules.rule(KeyMatch.inSet(Keys.SPACE), stopEventForFirefox) ] : [ ]),
  KeyRules.rule(KeyMatch.inSet(Keys.ESCAPE), specialInfo.onEscape)
];
```